frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -520,7 +522,13 @@ end
 
 function ReaderHighlight:onReadSettings(config)
     self.view.highlight.saved_drawer = config:readSetting("highlight_drawer") or self.view.highlight.saved_drawer
-    self.view.highlight.disabled = config:readSetting("highlight_disabled") or false
+    if config:readSetting("highlight_disabled")~=nil then
+        self.view.highlight.disabled = config:readSetting("highlight_disabled")
+    elseif G_reader_settings:readSetting("highlight_disabled")~=nil then
+        self.view.highlight.disabled = G_reader_settings:readSetting("highlight_disabled")
+    else
+        self.view.highlight.disabled = false
+    end
```

Comment:
Perhaps more efficient to write this block as below to reduce two function calls in the worst case ;)

```lua
local disable_hightlight = config:readSetting("highlight_disabled")
if disable_highlight == nil then
    disable_hightlight = G_reader_settings:readSetting("highlight_disabled") or false
end
self.view.highlight.disabled = disable_highlight
```